'make check' no longer works if configured with --disable-static

```
./configure --disable-static
make check
```
results in a build failure while linking ```tests/main```:
```
make[3]: Entering directory '/tmp/nghttp2-1.4.0/tests'
  CC       main.o
  CC       nghttp2_pq_test.o
  CC       nghttp2_map_test.o
  CC       nghttp2_queue_test.o
  CC       nghttp2_test_helper.o
  CC       nghttp2_frame_test.o
  CC       nghttp2_stream_test.o
  CC       nghttp2_session_test.o
  CC       nghttp2_hd_test.o
  CC       nghttp2_npn_test.o
  CC       nghttp2_helper_test.o
  CC       nghttp2_buf_test.o
  CCLD     main
main.o: In function `main':
/tmp/nghttp2-1.4.0/tests/main.c:54: undefined reference to `nghttp2_enable_strict_preface'
nghttp2_pq_test.o: In function `check_nodes':
/tmp/nghttp2-1.4.0/tests/nghttp2_pq_test.c:180: undefined reference to `nghttp2_pq_top'
/tmp/nghttp2-1.4.0/tests/nghttp2_pq_test.c:183: undefined reference to `nghttp2_pq_pop'
nghttp2_pq_test.o: In function `test_nghttp2_pq':
/tmp/nghttp2-1.4.0/tests/nghttp2_pq_test.c:61: undefined reference to `nghttp2_mem_default'
/tmp/nghttp2-1.4.0/tests/nghttp2_pq_test.c:61: undefined reference to `nghttp2_pq_init'
/tmp/nghttp2-1.4.0/tests/nghttp2_pq_test.c:62: undefined reference to `nghttp2_pq_empty'
/tmp/nghttp2-1.4.0/tests/nghttp2_pq_test.c:63: undefined reference to `nghttp2_pq_size'
[...]
```
This used to work prior to ce74a309.

--- a/configure.ac
+++ b/configure.ac
@@ -695,6 +695,9 @@ if test "x$threads" != "xyes" ||
     AC_DEFINE([NOTHREADS], [1], [Define to 1 if you want to disable threads.])
 fi
 
+# propagate $enable_static to tests/Makefile.am
+AM_CONDITIONAL([ENABLE_STATIC], [test "x$enable_static" = "xyes"])
+
 AC_SUBST([TESTLDADD])
 AC_SUBST([APPLDFLAGS])
 

--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -49,7 +49,15 @@ HFILES = nghttp2_pq_test.h nghttp2_map_test.h nghttp2_queue_test.h \
 
 main_SOURCES = $(HFILES) $(OBJECTS)
 
-main_LDADD = ${top_builddir}/lib/libnghttp2.la @CUNIT_LIBS@ @TESTLDADD@
+if ENABLE_STATIC
+main_LDADD = ${top_builddir}/lib/libnghttp2.la
+else
+# With static lib disabled and symbol hiding enabled, we have to link object
+# files directly because the tests use symbols not included in public API.
+main_LDADD = ${top_builddir}/lib/.libs/*.o
+endif
+
+main_LDADD += @CUNIT_LIBS@ @TESTLDADD@
 main_LDFLAGS = -static
 
 if ENABLE_FAILMALLOC

